src/remoted/syslogtcp.c

```diff
@@ -50,98 +73,23 @@ static void HandleClient(int client_socket, char *srcip)
 
     /* Initialize some variables */
     memset(buffer, '\0', OS_MAXSTR + 2);
-    memset(storage_buffer, '\0', OS_MAXSTR + 2);
-    memset(tmp_buffer, '\0', OS_MAXSTR + 2);
-
-
     while (1) {
         /* If an error occurred, or received 0 bytes, we need to return and close the socket */
-        r_sz = OS_RecvTCPBuffer(client_socket, buffer, OS_MAXSTR - 2);
+        r_sz = recv(client_socket, buffer + buff_size, (OS_MAXSTR - buff_size) - 2, 0);
```

Comment:
- `buffer` is `OS_MAXSTR + 2` bytes. Why do we read up to `OS_MAXSTR - 2`?
- `recv` does not null-terminate the buffer, as `OS_RecvTCPBuffer` does.

Should we use this?
```c
r_sz = recv(client_socket, buffer + buff_size, OS_MAXSTR - buff_size);
```